vizro-core/src/vizro/models/_dashboard.py

```diff
@@ -94,8 +96,10 @@ def pre_build(self):
         # Note redirect_from=["/"] doesn't work and so the / route must be defined separately.
         for order, page in enumerate(self.pages):
             path = page.path if order else "/"
-            dash.register_page(module=page.id, name=page.title, path=path, order=order, layout=page.build)
-        self._create_error_page_404()
+            dash.register_page(
+                module=page.id, name=page.title, path=path, order=order, layout=partial(self._make_page_layout, page)
+            )
+        dash.register_page(module=MODULE_PAGE_404, layout=create_layout_page_404())
```

Comment:
Putting this directly inline rather than having the separate `_create_error_page_404` is a small change but a big improvement to readability I think 👍 Very nice idea!

Just for consistency and ease of subclassing, let's rename `create_layout_page_404` to `_make_page_404_layout` and make it a `@staticmethod` inside `Dashboard` also I think.